devlib/module/sched.py

```diff
@@ -0,0 +1,239 @@
+#    Copyright 2018 Arm Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+
+from devlib.module import Module
+from devlib.utils.misc import memoized
+
+class SchedModule(Module):
+
+    name = 'sched'
+
+    sysfs_root = '/sys/devices/system/'
+
+    @staticmethod
+    def probe(target):
+        return True
```

Comment:
So the thing is it actually depends on a few different things... For capacities on Android, you'd usually want to have an energy model. On Mainline, it's just a device tree entry and is separate from having EAS. And then the config is not always exposed, but that's yet another different dependency...

The sched domains in general will always be exposed, so you'll always be able to e.g. check the sched domain flags.

Reply:
So *some* sort of test should definitely be here (you can query `target.os` to find out if target is running Android or Linux. It's up to you how through you wan to be with the test -- whether you want to check for everything or just a single key location which, if present, likely means the other stuff will be present too. There is a trade-off  here between the speed of the probe and risking missing something so that you end up failing much later than necessary.